docs/software_requirements/interrupts.sdoc

```diff
@@ -6,57 +6,228 @@ REQ_PREFIX: ZEP-
+  VALUE: ZEP-39
+
+[REQUIREMENT]
+UID: ZEP-13009
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Interrupts
+TITLE: Specific IRQ enable
+STATEMENT: >>>
+Zephyr RTOS shall provide a mechanism to Enable a specified IRQ.
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-39
+
+[REQUIREMENT]
+UID: ZEP-13010
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Interrupts
+TITLE: IRQ Enabled status
+STATEMENT: >>>
+Zephyr RTOS shall provide mechanisms that returns the Enabled status of a specified IRQ.
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-39
+
+[REQUIREMENT]
+UID: ZEP-13011
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Interrupts
+TITLE: ISR Context status
+STATEMENT: >>>
+Zephyr RTOS shall provide mechanisms that returns the execution context of In-ISR or Not In-ISR.
+<<<
+USER_STORY: >>>
+As the developer of functions that may run in either ISR or THREAD context, 
+I need to know the current context to enable condition behavior.
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-39
+
+[REQUIREMENT]
+UID: ZEP-13012
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Interrupts
+TITLE: Kernel Initialization status
+STATEMENT: >>>
+Zephyr RTOS shall provide mechanisms that returns the Kernel execution context 
+of Pre-Kernel, or In-Kernel.
 <<<
 USER_STORY: >>>
-As a Zephyr RTOS user I want to be able to enable each individual and a global interrupt according to my applications needs during program execution.
+As the developer of functions that may run either before or after Kernel initialization, 
+I need to know the current Kernel Initialization Status to enable condition behavior.
 <<<
+REVIEW_COMMENT: >>>
+Tim: Not relevant to the Interrupts requirements.  
+But I found it nearby and added it so it does not get lost.
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-39
 
 [REQUIREMENT]
-UID: ZEP-62
+UID: ZEP-13013
 STATUS: Draft
 TYPE: Functional
 COMPONENT: Interrupts
-TITLE: Disabling interrupts
+TITLE: Thread Preemptable status
 STATEMENT: >>>
-The Zephyr RTOS shall provide mechanisms to disable interrupts.
+Zephyr RTOS shall provide mechanisms that indicates if the 
+current execution context is a preemptable thread or not.
 <<<
 USER_STORY: >>>
-As a Zephyr RTOS user I want to be able to disable each individual and a global interrupt according to my applications needs during program execution.
+As the developer of functions that may run in either a preemptable thread or other context, 
+I need to know the current execution context to enable condition behavior.
+<<<
+REVIEW_COMMENT: >>>
+Tim: Not relevant to the Interrupts requirements.  
+But I found it nearby and added it so it does not get lost.
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-39
+
+[REQUIREMENT]
+UID: ZEP-13014
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Interrupts
+TITLE: Multi-level interrupts
+STATEMENT: >>>
+The Zephyr RTOS shall support multi-level preemptive interrupt priorities, when supported by hardware.
+Note: detailed analysis to demonstrate non interference will be needed here.
```

Comment:
Is this note needed? I think we should remove such notes and this should be proven by a test. Or did I get the note here wrong?

Reply:
I think you got the note correct.  I'll remove it.